patches/minecraft/net/minecraft/client/audio/SoundHandler.java.patch

```diff
@@ -1,14 +1,36 @@
 --- ../src-base/minecraft/net/minecraft/client/audio/SoundHandler.java
 +++ ../src-work/minecraft/net/minecraft/client/audio/SoundHandler.java
-@@ -67,6 +67,7 @@
+@@ -33,7 +33,7 @@
+ import org.apache.logging.log4j.Logger;
+ 
+ @SideOnly(Side.CLIENT)
+-public class SoundHandler implements IResourceManagerReloadListener, ITickable
++public class SoundHandler implements net.minecraftforge.client.resource.ISensitiveResourceReloadListener, ITickable
+ {
+     public static final Sound field_147700_a = new Sound("meta:missing_sound", 1.0F, 1.0F, 1, Sound.Type.FILE, false);
+     private static final Logger field_147698_b = LogManager.getLogger();
+@@ -63,15 +63,17 @@
+         this.field_147694_f = new SoundManager(this, p_i45122_2_);
+     }
+ 
+-    public void func_110549_a(IResourceManager p_110549_1_)
++    public void onResourceManagerReload(IResourceManager resourceManager, net.minecraftforge.client.resource.ReloadRequirements requirements)
      {
++        if (!requirements.shouldReload(net.minecraftforge.client.resource.VanillaResourceType.SOUNDS)) return;
          this.field_147697_e.func_148763_c();
  
+-        for (String s : p_110549_1_.func_135055_a())
 +        java.util.List<net.minecraft.util.Tuple<ResourceLocation, SoundList>> resources = new java.util.LinkedList<>();
-         for (String s : p_110549_1_.func_135055_a())
++        for (String s : resourceManager.func_135055_a())
```

Comment:
These parameter name changes shouldn't be here.

Add an entry in `forge.exc` for `onResourceManagerReload` so the `IResourceManager` parameter keeps its SRG name.

Reply:
Does it run for Forge files? I'm probably doing something wrong, but I can't seem to get that to work using
```
net/minecraftforge/client/resource/ISensitiveResourceReloadListener.onResourceManagerReload(Lnet/minecraft/client/resources/IResourceManager;Lnet/minecraftforge/client/resource/ReloadRequirements;)V=|p_110549_1_,requirements
```
Also, does `setupForge` have to be rerun after changing this?

Reply:
Ah right.
I think you need to add seperate lines for each vanilla class that's patched to implement that method.